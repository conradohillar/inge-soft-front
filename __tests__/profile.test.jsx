import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import Profile from '../app/(tabs)/profile';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { TamaguiProvider } from "tamagui";
import config from "../tamagui.config";
import axios from 'axios';

jest.mock('axios');

global.alert = jest.fn();

const queryClient = new QueryClient();
const Provider = ({ children }) => (
    <TamaguiProvider config={config}>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </TamaguiProvider>
);

describe('Profile Component', () => {

    const name = "Santiago Bassi";
    const email = "santaigobassi21@gmail.com";
    const url = "https://i.ibb.co/WxTJXCZ/de12caf4398b.jpg";
    const data = {
        "name": name,
        "email": email,
        "address": "Chascomús, Partido de Chascomús, Buenos Aires, Argentina",
        "dni": 123456789,
        "photo_url": url,
        "is_driver": true
    };

    axios.get.mockResolvedValue({ data });

    it('data ok', async () => {
        const { getByText, findByTestId } = render(<Provider><Profile /></Provider>);

        await waitFor(() => {
            expect(getByText(name)).toBeTruthy();
            expect(getByText(email)).toBeTruthy();
            expect(findByTestId("profile-picture")).toBeTruthy();
            expect(getByText("Credenciales")).toBeTruthy();
            expect(getByText("Mis autos")).toBeTruthy();
        });
    });



    describe('profile page Snapshot', () => {
        it('debería coincidir con la snapshot', () => {
            const tree = render(<Provider><Profile /></Provider>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

});