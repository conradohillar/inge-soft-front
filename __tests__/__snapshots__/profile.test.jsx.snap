// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component profile page Snapshot deber√≠a coincidir con la snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 2,
        "borderStyle": "solid",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": "8%",
        "paddingLeft": 16,
        "paddingRight": 16,
        "width": "100%",
      }
    }
  >
    <View
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 40,
          "justifyContent": "center",
          "paddingLeft": 18,
          "paddingRight": 18,
          "width": 40,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          [
            {
              "height": 32,
            },
            {
              "width": 32,
            },
          ]
        }
      />
    </View>
    <Text
      href="/home"
      onPress={[Function]}
      role="link"
      style={
        [
          {
            "fontSize": 48,
            "lineHeight": 48,
          },
          {
            "fontFamily": "Quicksand-Bold",
          },
          {
            "color": "#59A58A",
          },
          {
            "marginTop": 8,
          },
        ]
      }
    >
      rydio
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "15%",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "90%",
          }
        }
      >
        <View
          style={
            [
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 100000,
                  "borderBottomRightRadius": 100000,
                  "borderBottomWidth": 1,
                  "borderLeftWidth": 1,
                  "borderRightWidth": 1,
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 100000,
                  "borderTopRightRadius": 100000,
                  "borderTopWidth": 1,
                  "flexDirection": "column",
                  "height": 144,
                  "justifyContent": "center",
                  "maxHeight": 144,
                  "maxWidth": 144,
                  "minHeight": 144,
                  "minWidth": 144,
                  "overflow": "hidden",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "position": "relative",
                  "width": 144,
                }
              }
            >
              <View
                style={
                  {
                    "bottom": 0,
                    "flexDirection": "column",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 1,
                  }
                }
              >
                <Image
                  accessibilityLabel="Cam"
                  fullscreen={true}
                  onError={[Function]}
                  onLoad={[Function]}
                  source={
                    {
                      "uri": "https://i.ibb.co/WxTJXCZ/de12caf4398b.jpg",
                    }
                  }
                  style={
                    {
                      "height": 144,
                      "width": 144,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "column",
              "justifyContent": "space-evenly",
              "marginLeft": 20,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000",
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "Quicksand-Bold",
                  },
                ]
              }
            >
              Santiago Bassi
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  [
                    {
                      "height": 16,
                    },
                    {
                      "width": 16,
                    },
                  ]
                }
                tintColor="#aaa"
              />
            </View>
          </View>
          <Text
            style={
              [
                {
                  "color": "#4b5563",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "fontFamily": "Quicksand-SemiBold",
                },
              ]
            }
          >
            santaigobassi21@gmail.com
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "column",
          "height": "70%",
          "width": "100%",
        }
      }
    >
      <View
        borderTopColor="#ddd"
        borderTopWidth={2}
        style={
          [
            {
              "width": "100%",
            },
            {
              "height": "20%",
            },
            {
              "alignItems": "center",
            },
            {
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": "80%",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              [
                {
                  "height": 24,
                },
                {
                  "width": 24,
                },
              ]
            }
            tintColor="#aaa"
          />
          <Text
            href="/MyCarsPage"
            onPress={[Function]}
            role="link"
            style={
              [
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
                {
                  "color": "#000",
                },
                {
                  "fontFamily": "Quicksand-Bold",
                },
                [
                  {
                    "marginLeft": 20,
                  },
                ],
              ]
            }
          >
            Mis autos
          </Text>
        </View>
      </View>
      <View
        borderTopColor="#ddd"
        borderTopWidth={2}
        style={
          [
            {
              "width": "100%",
            },
            {
              "height": "20%",
            },
            {
              "alignItems": "center",
            },
            {
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": "80%",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              [
                {
                  "height": 24,
                },
                {
                  "width": 24,
                },
              ]
            }
            tintColor="#aaa"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginLeft": 20,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontSize": 20,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#000",
                  },
                  {
                    "fontFamily": "Quicksand-Bold",
                  },
                ]
              }
            >
              Credenciales
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
